src/main/java/io/confluent/examples/streams/JsonToAvroExample.java

```diff
@@ -0,0 +1,133 @@
+ * }</pre>
+ * Note: The above commands are for the Confluent Platform. For Apache Kafka it should be {@code bin/kafka-topics.sh ...}.
+ * <p>
+ * 3) Start this example application either in your IDE or on the command line.
+ * <p>
+ * If via the command line please refer to <a href='https://github.com/confluentinc/kafka-streams-examples#packaging-and-running'>Packaging</a>.
+ * Once packaged you can then run:
+ * <pre>
+ * {@code
+ * $ java -cp target/kafka-streams-examples-5.2.0-SNAPSHOT-standalone.jar io.confluent.examples.streams.JsonToAvroExample
+ * }
+ * </pre>
+ * 4) Write some input data to the source topics (e.g. via {@link JsonToAvroExampleDriver}).
+ * The already running example application (step 3) will automatically process this input data and
+ * write the results to the output topic. The {@link JsonToAvroExampleDriver} will print the
+ * results from the output topic
+ * <pre>
+ * {@code
+ * # Here: Write input data using the example driver.  Once the driver has stopped generating data,
+ * # you can terminate it via Ctrl-C.
+ * $ java -cp target/kafka-streams-examples-5.2.0-SNAPSHOT-standalone.jar io.confluent.examples.streams.JsonToAvroExampleDriver
+ * }
+ * </pre>
+ */
+public class JsonToAvroExample {
+
+  static final String JSON_SOURCE_TOPIC = "json-source";
+  static final String AVRO_SINK_TOPIC = "avro-sink";
+
+  public static void main(final String[] args) {
+    final String bootstrapServers = args.length > 0 ? args[0] : "localhost:9092";
+    final String schemaRegistryUrl = args.length > 1 ? args[1] : "http://localhost:8081";
+    final KafkaStreams streams = buildJsonToAvroStream(
+        bootstrapServers,
+        schemaRegistryUrl
+    );
+    streams.start();
+
+    // Add shutdown hook to respond to SIGTERM and gracefully close Kafka Streams
+    Runtime.getRuntime().addShutdownHook(new Thread(streams::close));
+  }
+
+  static KafkaStreams buildJsonToAvroStream(final String bootstrapServers,
+      final String schemaRegistryUrl) {
+    final Properties streamsConfiguration = new Properties();
+
+    streamsConfiguration.put(StreamsConfig.APPLICATION_ID_CONFIG, "json-to-avro-stream-conversion");
+    streamsConfiguration.put(StreamsConfig.CLIENT_ID_CONFIG, "json-to-avro-stream-conversion-client");
+    streamsConfiguration.put(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapServers);
+    // Where to find the Confluent schema registry instance(s)
+    streamsConfiguration.put(AbstractKafkaAvroSerDeConfig.SCHEMA_REGISTRY_URL_CONFIG, schemaRegistryUrl);
+
+    streamsConfiguration.put(StreamsConfig.DEFAULT_VALUE_SERDE_CLASS_CONFIG, SpecificAvroSerde.class);
+    streamsConfiguration.put(StreamsConfig.DEFAULT_KEY_SERDE_CLASS_CONFIG, Serdes.String().getClass().getName());
+    streamsConfiguration.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+    streamsConfiguration.put(StreamsConfig.COMMIT_INTERVAL_MS_CONFIG, 10 * 1000);
```

Comment:
Should we maybe set this to 100ms for an interactive example?

Reply:
ack